src/ClientBuilder.php

```diff
@@ -314,6 +315,17 @@ private function createTransportInstance(): TransportInterface
 
         $this->messageFactory = $this->messageFactory ?? MessageFactoryDiscovery::find();
         $this->uriFactory = $this->uriFactory ?? UriFactoryDiscovery::find();
+
+        if (null === $this->httpClient && null !== $this->options->getHttpProxy()) {
```

Comment:
We should make this fail hard if a client is already in, or the built client will be silently without proxy.